### PR TITLE
chore: handle exception and page converter

### PR DESCRIPTION
**Pull request by nvagno**

### CODE REVIEW

File: src/main/java/com/blogify/blogapi/endpoint/rest/InternalToRestExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    log.error("Internal error", e);
+    return new ResponseEntity<>(
+        toRest(e, HttpStatus.INTERNAL_SERVER_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+  }
+
+  private Exception toRest(java.lang.Exception e, HttpStatus status) {
+    var restException = new Exception();
+    // TODO : should be Exception generated by Openapi (My IDE does not index correctly)
+    // restException.setType(status.toString());
+    // restException.setMessage(e.getMessage());
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/endpoint/rest/InternalToRestExceptionHandler.java:45]: first version of spec APi is finish, u can use it now

File: src/main/java/com/blogify/blogapi/model/BoundedPageSize.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Getter;
+
+@Getter
+public class BoundedPageSize {
+  public static final int MAX_SIZE = 500;
+  private final int value;
+
+  public BoundedPageSize(int value) {
+    if (value < 1) {
+      throw new BadRequestException("page size must be >=1");
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/model/BoundedPageSize.java]: Start message with uppercase, finish with point

File: src/main/java/com/blogify/blogapi/model/BoundedPageSize.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    if (value > MAX_SIZE) {
+      throw new BadRequestException("page size must be <" + MAX_SIZE);
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/model/BoundedPageSize.java]: same

File: src/main/java/com/blogify/blogapi/model/PageFromOne.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Getter;
+
+@Getter
+public class PageFromOne {
+  public static final int MIN_PAGE = 1;
+  private final int value;
+
+  public PageFromOne(int value) {
+    if (value < MIN_PAGE) {
+      throw new BadRequestException("page must be >=1");
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/model/PageFromOne.java]: Same

File: src/main/java/com/blogify/blogapi/model/exception/ApiException.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.blogify.blogapi.model.exception;
+
+public class ApiException extends RuntimeException {
+  private final ExceptionType type;
```
Comment (heiherilala) [inline src/main/java/com/blogify/blogapi/model/exception/ApiException.java:4]: when u make it final, can u use, this.type = type?

**Review by heiherilala — CHANGES_REQUESTED** [review]

File: src/main/java/com/blogify/blogapi/model/exception/ApiException.java (hunk 5 above)
Comment (YumeT023) [reply src/main/java/com/blogify/blogapi/model/exception/ApiException.java:4]: You can assign it _INITIALIZE_ it _ONCE_ in the constructor
so, yeah you can this.type = type in the constructor

**Pull request merged**